### PR TITLE
[Go] Scope struct tags as annotations

### PR DESCRIPTION
**Pull request by keith-hall**
Scope Golang's [struct tags](https://pkg.go.dev/reflect#StructTag) as annotations.
Further reading:
- https://www.digitalocean.com/community/tutorials/how-to-use-struct-tags-in-go
- https://stackoverflow.com/a/30889373/4473405

~~I have kept the original `string.quoted.backtick` scoping as well for backwards compatibility with users whom might prefer these to stay highlighted as before.~~

### CODE REVIEW

File: Go/Go.sublime-syntax [hunk 1]
```diff
@@ -660,18 +660,20 @@ contexts:
...
-          pop: 1
-        - include: match-template-string
-        - match: '{{char_escape}}'
-          scope: constant.character.escape.go
-        - match: \%%
-          scope: constant.character.escape.go
-        - include: match-fmt
+      push: inside-interpreted-string
+
+  inside-interpreted-string:
```
[the remark this answers is not in the document]
Comment (keith-hall) [reply Go/Go.sublime-syntax]: agreed, done :)

File: Go/Go.sublime-syntax [hunk 2]
```diff
@@ -1257,7 +1259,36 @@ contexts:
...
+          captures:
+            1: meta.annotation.identifier.go variable.annotation.go
+            2: meta.annotation.go punctuation.separator.key-value.go
+          push:
+            - clear_scopes: 1
+            - match: '"'
+              scope: punctuation.definition.string.begin.go
+              push:
+                - meta_include_prototype: false
+                - meta_scope: string.quoted.double.go meta.annotation.parameters.go
```
[the remark this answers is not in the document]
Comment (keith-hall) [reply Go/Go.sublime-syntax]: yup, done :)

File: Go/Go.sublime-syntax [hunk 3]
```diff
@@ -1257,7 +1259,36 @@ contexts:
       scope: variable.other.member.declaration.go
       push: pop-struct-field-meta
 
-    - include: match-any
+    - include: struct-tag-annotation
+    - include: match-semicolon
+    - include: match-comma
+    - include: match-operators
+
+  struct-tag-annotation:
```
[the remark this answers is not in the document]
Comment (keith-hall) [reply Go/Go.sublime-syntax]: fixed

File: Go/Go.sublime-syntax [hunk 4]
```diff
@@ -1257,7 +1259,36 @@ contexts:
...
+    - match: '`'
+      scope: punctuation.definition.string.begin.go punctuation.definition.annotation.begin.go
+      push:
+        - meta_include_prototype: false
+        - meta_scope: string.quoted.backtick.go meta.annotation.go
```
[the remark this answers is not in the document]
Comment (keith-hall) [reply Go/Go.sublime-syntax]: removed the `string.quoted.backtick` scoping of these struct tag annotations :)

File: Go/Go.sublime-syntax [hunk 5]
```diff
@@ -1257,7 +1259,36 @@ contexts:
...
+        - match: '`'
+          scope: punctuation.definition.string.end.go punctuation.definition.annotation.end.go
+          pop: 1
+        - match: (\w+)(:)
+          captures:
+            1: meta.annotation.identifier.go variable.annotation.go
+            2: meta.annotation.go punctuation.separator.key-value.go
+          push:
```
[the remark this answers is not in the document]
Comment (keith-hall) [reply Go/Go.sublime-syntax]: moved to named contexts

**Review by deathaxe — APPROVED** [review]

**Comment by TerminalFi** [comment]
@deathaxe @keith-hall I've been using this for 2 weeks now.  This is a great addition!

**Review by michaelblyons — APPROVED** [review]

**Pull request merged**